internal/clients/sources/sources_impl.go

```diff
@@ -2,14 +2,69 @@ package sources
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 
+	"github.com/RHEnVision/provisioning-backend/internal/cache"
 	"github.com/RHEnVision/provisioning-backend/internal/config"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/RHEnVision/provisioning-backend/internal/parsing"
 )
 
+type AppType struct {
+	Id          string `json:"id"`
+	Name        string `json:"name"`
+	DisplayName string `json:"display_name"`
+}
+
+type dataElement struct {
+	Data []AppType `json:"data"`
+}
+
 func getSourcesClient(ctx context.Context) (SourcesIntegration, error) {
 	return NewClientWithResponses(config.Sources.URL)
 }
 
 func init() {
 	GetSourcesClient = getSourcesClient
 }
+
+func (c *ClientWithResponses) GetProvisioningTypeId(ctx context.Context, reqEditors ...RequestEditorFn) (string, error) {
+	if appTypeId, ok := cache.Cache.Get(cache.AppTypeKey); ok {
+		return appTypeId.(string), nil
+	}
+	if err := loadAppId(ctx, c); err != nil {
+		return "", err
+	}
+	if appTypeId, ok := cache.Cache.Get(cache.AppTypeKey); ok {
+		return appTypeId.(string), nil
```

Comment:
A function cache.GetAppTypeKey would be really beneficial because it would hide the implementation, in Go 1.18 Generics could further simplify this.

Reply:
yeah I was thinking about that, but then realized it should always be string so it is kinda fine... but I can implement that if you feel it's better :)

Reply:
I'd rather not in this PR if it's fine with you